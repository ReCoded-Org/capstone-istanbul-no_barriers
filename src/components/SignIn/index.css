:root {
  --title-color: #002736;
  --subTitle-color: #000000;
  --background-color: #fc9e01;
  --text-color: #eff4f6;
  --filter-shadow-color: #b3b3b3;
  --link-color: #0641af;
}

.signIn {
  padding: 3rem;
}

.icons button {
  border-radius: 50%;
  margin: 1rem;
}

.signInButton {
  background-color: var(--background-color);
  color: white;
  font-weight: 600;
  cursor: pointer;
  width: 7rem;
  height: 3rem;
  font-size: 1rem;
  border-radius: 0.5rem;
  border-style: hidden;
  box-shadow: 0px 11px 24px -7px var(--filter-shadow-color);
}

.socialIcons {
  background-color: var(--background-color);
  border: 0rem;
  box-shadow: 0px 11px 24px -7px var(--filter-shadow-color);
}

.reset {
  color: var(--background-color);
  font-weight: bold;
  cursor: pointer;
}
