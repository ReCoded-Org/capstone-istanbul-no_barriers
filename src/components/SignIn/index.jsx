import * as firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import Row from "react-bootstrap/Row";
import Container from "react-bootstrap/Container";
import Col from "react-bootstrap/Col";
import Alert from "react-bootstrap/Alert";
import "./index.css";
import { useTranslation } from "react-i18next";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faGoogle } from "@fortawesome/free-brands-svg-icons";
import { faFacebook } from "@fortawesome/free-brands-svg-icons";

const SignIn = () => {
  const history = useHistory();
  const [alert, setAlert] = useState(null);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const googleProvider = new firebase.auth.GoogleAuthProvider();
  const facebookProvider = new firebase.auth.FacebookAuthProvider();

  const handleFacebookLogin = (e) => {
    firebase
      .auth()
      .signInWithPopup(facebookProvider)
      .then(function (result) {
        setAlert({
          message: "signIn.successMsg",
          status: "success",
        });
        history.push("/");
      })
      .catch(function (error) {
        setAlert({
          message: error.message,
          status: "danger",
        });
      });
  };

  const handleGoogleLogin = (e) => {
    firebase
      .auth()
      .signInWithPopup(googleProvider)
      .then(function (result) {
        setAlert({
          message: "signIn.successMsg",
          status: "success",
        });
        history.push("/");
      })
      .catch(function (error) {
        setAlert({
          message: error.message,
          status: "danger",
        });
      });
  };

  const handleInput = (e) => {
    const value = e.target.value;
    if (e.target.type === "email") {
      setEmail(value);
    } else if (e.target.type === "password") {
      setPassword(value);
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then((user) => {
        setAlert({
          message: "signIn.successMsg",
          status: "success",
        });
        history.push("/");
      })
      .catch(function (error) {
        setAlert({
          message: "signIn.errorMsg",
          status: "danger",
        });
      });
  };
  const { t } = useTranslation();

  return (
    <Container className="container justify-content-center">
      <Row className="justify-content-center">
        <h4 className="welcomeBack">{t("signIn.welcomeBack")}</h4>
      </Row>
      <Row className="justify-content-center">
        <p>{t("signIn.signInText")}</p>
      </Row>
      <Row className="justify-content-center">
        <div className="icons">
          <Button
            className="socialIcons"
            onClick={handleGoogleLogin}
            variant="danger"
          >
            <FontAwesomeIcon icon={faGoogle} />
          </Button>
          <Button className="socialIcons" onClick={handleFacebookLogin}>
            <FontAwesomeIcon icon={faFacebook} />
          </Button>
        </div>
      </Row>
      <Form id="signInForm" onSubmit={handleSubmit}>
        <Form.Group as={Row} className="justify-content-center">
          <p>{t("signIn.or")}</p>
        </Form.Group>
        <Form.Group
          as={Row}
          className="justify-content-center"
          controlId="formBasicEmail"
        >
          <Col xs={5}>
            <Form.Control
              onChange={handleInput}
              type="email"
              placeholder={t("signIn.enterEmail")}
            />
            <Form.Text className="text-muted">{t("signIn.emailMsg")}</Form.Text>
          </Col>
        </Form.Group>

        <Form.Group
          as={Row}
          className="justify-content-center"
          controlId="formBasicPassword"
        >
          <Col xs={5}>
            <Form.Control
              onChange={handleInput}
              type="password"
              placeholder={t("signIn.password")}
            />
          </Col>
        </Form.Group>
        <Row className="justify-content-center">
          {alert && <Alert variant={alert.status}>{t(alert.message)}</Alert>}
        </Row>
        <Row className="justify-content-center">
          <Button variant="primary" type="submit" className="signInButton">
            {t("signIn.signInBtn")}
          </Button>
        </Row>
      </Form>
    </Container>
  );
};

export default SignIn;
