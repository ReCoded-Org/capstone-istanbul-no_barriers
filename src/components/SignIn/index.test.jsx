import React from "react";
import renderer from "react-test-renderer";
import SignIn from "./index";

it("SignIn renders correctly", () => {
  const tree = renderer.create(<SignIn />).toJSON();
  expect(tree).toMatchSnapshot();
});
