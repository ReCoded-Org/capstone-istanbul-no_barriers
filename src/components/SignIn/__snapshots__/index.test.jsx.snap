// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly 1`] = `
<div
  className="justify-content-center signIn container"
>
  <div
    className="justify-content-center row"
  >
    <h4
      className="welcomeBack"
    >
      signIn.welcomeBack
    </h4>
  </div>
  <div
    className="justify-content-center row"
  >
    <p>
      signIn.signInText
    </p>
  </div>
  <div
    className="justify-content-center row"
  >
    <div
      className="icons"
    >
      <button
        className="socialIcons btn btn-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 "
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <button
        className="socialIcons btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-facebook fa-w-16 "
          data-icon="facebook"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <form
    className=""
    id="signInForm"
    onSubmit={[Function]}
  >
    <div
      className="justify-content-center form-group row"
    >
      <p>
        signIn.or
      </p>
    </div>
    <div
      className="justify-content-center form-group row"
    >
      <div
        className="col-5"
      >
        <input
          className="form-control"
          id="formBasicEmail"
          onChange={[Function]}
          placeholder="signIn.enterEmail"
          type="email"
        />
        <small
          className="text-muted form-text"
        >
          signIn.emailMsg
        </small>
      </div>
    </div>
    <div
      className="justify-content-center form-group row"
    >
      <div
        className="col-5"
      >
        <input
          className="form-control"
          id="formBasicPassword"
          onChange={[Function]}
          placeholder="signIn.password"
          type="password"
        />
      </div>
    </div>
    <div
      className="justify-content-center row"
    />
    <div
      className="justify-content-center row"
    >
      <button
        className="signInButton btn btn-primary"
        disabled={false}
        type="submit"
      >
        signIn.signInBtn
      </button>
    </div>
  </form>
</div>
`;
